client/components/Signup.js

```diff
@@ -40,59 +70,92 @@ class Signup extends Component {
                 email: '',
                 password: ''
             },
-            dirty: false
+            dirtyPassword: false,
+            dirtyEmail: false
         })
-
+        this.props.clearCheckAvail()
     }
 
     render() {
         return (
             <div>
-                <form onSubmit={this.handleSubmit}>
-                    <label>Username: </label>
-                    <input
-                        name='username'
-                        type='text'
-                        value={this.state.account.username}
-                        onChange={this.handleChange}
-                        required
-                    />
-                    <label>Email: </label>
-                    <input
-                        name='email'
-                        type='text'
-                        value={this.state.account.email}
-                        onChange={this.handleChange}
-                        required
-                    />
-                    <label>Password: </label>
-                    <small>(Must be at least 6 characters long)</small>
-                    <input
-                        name='password'
-                        type='password'
-                        value={this.state.account.password}
-                        onChange={this.handleChange}
-                        required
-                    />
-                    {this.state.dirty && (this.state.account.password.length < 6 || this.state.account.password.length > 50) ? (<p>Invalid password</p>) : null}
-                    <button type='submit' disabled={(this.state.account.password.length < 6) || (this.state.account.password.length > 50)}>Create Account!</button>
-                </form>
-                <Google />
+                <Center>
+                    <form onSubmit={this.handleSubmit}>
+                        <TextField
+                            name='username'
+                            floatingLabelText='Username'
+                            type='input'
+                            value={this.state.account.username}
+                            onChange={this.handleChange}
+                            errorStyle={(this.props.message && this.props.message==='Username available') ? {color: 'green'} : {color: 'red'}}
```

Comment:
Your error messages need to be a little more descriptive.  If the login failed because the user already exists, don't tell them "user might already exist".  Instead, say something like, "An account with this email already exists.  Log in to your account." or something....

Reply:
for security reasons, the error message is deliberately vague. you can just keep trying to sign up with taken emails to see what accounts exist otherwise and then bruteforce the corresponding passwords.